src/plugins/data/public/ui/dataset_selector/configurator.tsx

```diff
@@ -112,15 +138,16 @@ export const Configurator = ({
                     text: field.displayName || field.name,
                     value: field.name,
                   })),
-                  { text: '-----', value: '', disabled: true },
-                  { text: 'No time field', value: undefined },
+                  { text: '-----', value: '-----', disabled: true },
+                  { text: noTimeFilter, value: noTimeFilter },
```

Comment:
oh this is weird. so undefined never worked? or was it because we were trying to compare to a string `undefined` ?

Reply:
`OuiSelectable` has some weird properties when attempting to match an `undefined` value prop with what's in the array. Made it easier to just set the values as what the text is and change it in the `onChange`.